src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -866,10 +865,18 @@ public Object eval(Object[] args, EvaluationContext ec) {
                                 // the situations where that context would be used don't make sense for trying to reverse a
                                 // select value back to a label in an unrelated expression
                                 if (ref.isAmbiguous()) {
-                                    // SurveyCTO: We need a absolute "ref" to populate the dynamic choices,
-                                    // like we do when we populate those at FormEntryPrompt (line 251).
-                                    // The "ref" here is ambiguous, so we need to make it concrete first.
+                                    // ref, the reference used to specify where choices are defined, could be an absolute
```

Comment:
This `IFunctionHandler` implementation should be moved out to another file now we're touching it again.